crates/solvers/src/boundary/legacy.rs

```diff
@@ -544,6 +552,30 @@ fn to_domain_solution(
     })
 }
 
+fn to_boundary_auction_result(notification: &notification::Notification) -> (i64, AuctionResult) {
+    let auction_id = match notification.auction_id {
+        auction::Id::Solve(id) => id,
+        auction::Id::Quote => 0,
+    };
+
+    let auction_result = match notification.kind {
+        notification::Kind::EmptySolution => {
+            AuctionResult::Rejected(SolverRejectionReason::NoUserOrders)
+        }
+        notification::Kind::ScoringFailed => {
+            AuctionResult::Rejected(SolverRejectionReason::NonPositiveScore)
+        }
+        notification::Kind::NonBufferableTokensUsed => AuctionResult::Rejected(
+            SolverRejectionReason::NonBufferableTokensUsed(Default::default()),
```

Comment:
can we drag the token set here for better debugability

Reply:
Added.